smart-contracts/contracts/cl-vault/src/test_tube/proptest.rs

```diff
@@ -0,0 +1,430 @@
+                    address: account.address(),
+                    denom: denom.to_string()
+                }
+            )
+            .unwrap()
+    }
+
+    fn get_user_shares_balance(
+        wasm: &Wasm<OsmosisTestApp>,
+        contract_address: &Addr,
+        account: &SigningAccount,
+    ) -> UserBalanceResponse {
+        wasm
+            .query(
+            contract_address.as_str(),
+            &QueryMsg::VaultExtension(ExtensionQueryMsg::Balances(
+                crate::msg::UserBalanceQueryMsg::UserSharesBalance {
+                    user: account.address(),
+                },
+            )),
+        )
+        .unwrap()
+    }
+
+    fn get_position_assets(
+        wasm: &Wasm<OsmosisTestApp>,
+        contract_address: &Addr,
+    ) -> TotalAssetsResponse {
+        wasm
+            .query(
+            contract_address.as_str(),
+            &QueryMsg::TotalAssets {},
+        )
+        .unwrap()
+    }
+
+    fn get_position_ticks(
+        wasm: &Wasm<OsmosisTestApp>,
+        cl: &ConcentratedLiquidity<OsmosisTestApp>,
+        contract_address: &Addr,
+    ) -> (i64, i64) {
+        // query_position will return a Vec of position_ids
+        let position_response: PositionResponse = wasm
+            .query(
+                contract_address.as_str(),
+                &QueryMsg::VaultExtension(ExtensionQueryMsg::ConcentratedLiquidity(
+                    crate::msg::ClQueryMsg::Position {},
+                )),
+            )
+            .unwrap();
+
+        // TODO Use those to take the latest one? or what?
+        let position = cl
+            .query_position_by_id(
+                &PositionByIdRequest {
+                    position_id: position_response.position_ids[0]
+                }
+            )
+            .unwrap().position.unwrap().position;
+
+        match position {
+            Some(position) => (position.lower_tick, position.upper_tick),
+            None => panic!("Position not found"),
+        }
+    }
+
+    // ASSERT METHODS
+
+    fn assert_deposit_withdraw(
+        wasm: &Wasm<OsmosisTestApp>,
+        contract_address: &Addr,
+        accounts: &Vec<SigningAccount>,
+        accounts_shares_balance: &HashMap<String, Uint128>,
+    ) {
+        // TODO: multi-query foreach user created previously
+        for account in accounts {
+            let shares = get_user_shares_balance(wasm, contract_address, account);
+
+            // Check that the current account iterated shares balance is the same we expect from Hashmap
+            //assert_eq!(shares.balance, accounts_shares_balance.get(&account.address()));
+        }
+    }
+
+    fn assert_swap() {
+        todo!()
+    }
+
+    fn assert_update_range() {
+        todo!()
+    }
+
+    // COMPOSE STRATEGY
+
+    // get_initial_range generates random lower and upper ticks for the initial position
+    prop_compose! {
+        // TODO: evaluate if lower_tick and upper_tick are too much arbitrary
```

Comment:
This can be evaluated against osmosis docs, from the [osmosis source code](https://github.com/osmosis-labs/osmosis/blob/main/x/concentrated-liquidity/types/constants.go#L22) 
```go
MinInitializedTick, MaxTick int64 = -108000000, 342000000
````